### PR TITLE
Create topics onboarding page

### PR DESCRIPTION
**Pull request by jackcrane**
Created the slide in the onboarding process to allow selection of applicable tags.

### CODE REVIEW

**Comment by jackcrane** [comment]
After a very productive huddle with @calebpanza we reorganized the code to better match Apollos conventions and to utilize the `ui-kit`. This is still using dummy data in `templates/mobile/src/onboarding/slides/TagGroups/TagGroupsConnected.js` and has no reactivity or webcode yet, but progress nonetheless! Attached is a screenshot of the current state.

Screen Shot 2022-11-21 at 11 45 54 PM

**Comment by calebpanza** [comment]
🙌🙌🙌

**Comment by jackcrane** [comment]
In [810971e](https://github.com/ApollosProject/apollos-apps/pull/2799/commits/810971e7277b482981512489acb4a00461e85f3d) I pulled the `TagContainer` that holds the `PressableChip` into its own component so it can manage its own state and make requests to the API on selection/deselection. The API is not completely caught up so I cannot completely test the integration but the bones are here :). I also set up a bunch of test data in the staging db (`tag_group_tags`)

Screen Shot 2022-11-23 at 4 53 04 PM

**Comment by jackcrane** [comment]
Finalizing the feature addition. Final commits caught the UI up to master and incorporated error handling. UI Appearance has not changed since the screenshot above, and attached is a screen recording of network failure behavior.

https://user-images.githubusercontent.com/45675751/208980704-7d6cfa33-6c43-470d-803f-4adc1e194589.mov

File: templates/mobile/src/components/TagGroup/TagGroup.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  tags: PropTypes.arrayOf(PropTypes.shape({})),
+};
+TagGroup.defaultProps = {
+  title: 'hgdf',
+  tags: [],
+};
+
+export default TagGroup;
+
+const Tag = ({ tag, i }) => {
```
Comment (calebpanza) [inline templates/mobile/src/components/TagGroup/TagGroup.js:43]: Could you break out these components into their own files? Will make it easier to run tests later on and easier to track down the location.

**Review by calebpanza — CHANGES_REQUESTED** [review]

**Comment by jackcrane** [comment]
I have added all the tests I can think to do, have tested the functionality as much as I reasonably can. I will continue brainstorming ideas to better test the components and improve coverage. 

The naming is slightly confusing with singular and plural `TagGroup` and `TagGroups`, so below is a hierarchy

```javascript
<TagGroups> // covered by TagGroups.test.js
  <TagGroup> // covered by TagGroup.test.js
    <Tag /> // covered by Tag.test.js
    /* ... any number of Tag components ... */
  </TagGroup>
  /* ... any number of TagGroup components ... */
</TagGroups>
```

Screen Shot 2023-01-06 at 11 26 44 AM

I would appreciate some input on additional tests to add to the suite, especially on `TagGroup.js`, covered by `TagGroup.test.js`.

File: templates/mobile/src/components/TagGroup/TagGroup.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    </Styled>
+  );
+};
+
+TagGroup.propTypes = {
+  title: PropTypes.string,
+  tags: PropTypes.arrayOf(PropTypes.shape({})),
+};
+TagGroup.defaultProps = {
+  title: 'hgdf',
```
Comment (dsilva609) [inline templates/mobile/src/components/TagGroup/TagGroup.js:32]: Should this title just be an empty string for default?

File: templates/mobile/src/onboarding/slides/TagGroups/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import TagGroups from './TagGroups';
+import TagGroupsConnected from './TagGroupsConnected';
+
+// export FollowConnected from './FollowConnected';
```
Comment (dsilva609) [inline templates/mobile/src/onboarding/slides/TagGroups/index.js:4]: is this needed?

**Review by dsilva609 — CHANGES_REQUESTED** [review]

File: templates/mobile/src/components/TagGroup/TagGroup.js (hunk 2 above)
Comment (jackcrane) [reply templates/mobile/src/components/TagGroup/TagGroup.js:32]: Yes, will fix

**Review by calebpanza — CHANGES_REQUESTED** [review]
CleanShot 2023-01-17 at 12 38 31

Default behavior when there is _no_ tag groups coming from the server should be to remove this screen from onboarding.

**Comment by calebpanza** [comment]
@jackcrane looks like there is a missing import on the `useTagSelected` hook.

CleanShot 2023-01-17 at 13 07 16

For that hook, could you make sure to have that mutation localized inside of `useTagGroup` and not to have `SELECT_TAG` be imported?

Could we also move that hook into the `utils` directory? (I plan to rename that directory to `hooks` during cooldown)

Lastly, let's get some level of test coverage on that hook in order to avoid any future misgivings.

File: templates/mobile/src/components/TagGroup/useTagSelected.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { useMutation } from '@apollo/client';
+import { SELECT_TAG } from '../../onboarding/slides/TagGroups/getTagGroups.js';
+
+export const useTagSelected = () => {
+  const [selectionMutationRequest, { loading, error }] =
+    useMutation(SELECT_TAG);
+
+  const selectTag = (tagId) => {
```
Comment (dsilva609) [inline templates/mobile/src/components/TagGroup/useTagSelected.js:8]: Just a small thing on the naming on this variable. Seeing `select` here makes it seem like what's being done is a query. Would changing it to something like `updateTagSelection` convey the same meaning while also sounding more like a mutation is happening?

File: templates/mobile/src/onboarding/slides/TagGroups/selectTag.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { gql } from '@apollo/client';
+
+export default gql`
+  mutation selectTag($tagId: ID) {
+    selectTag(tagId: $tagId) {
```
Comment (dsilva609) [inline templates/mobile/src/onboarding/slides/TagGroups/selectTag.js:5]: Echoing the comment from above, but could this be renamed to something like `updateTagSelection`?

File: templates/mobile/src/components/TagGroup/useTagSelected.js (hunk 4 above)
Comment (calebpanza) [reply templates/mobile/src/components/TagGroup/useTagSelected.js:8]: or rather `useUpdateTagSelection` to follow in proper hook naming standards

**Comment by calebpanza** [comment]
@jackcrane do you need help getting this landed? I can make some time to dive into this if you need some help. Would really like to get this landed in the next few days

**Comment by jackcrane** [comment]
i should be able to get it there over the weekend. right now all that is left is to add tests for the hook. 

as of now i cannot get the app to launch in sim on my mac which has held some of my progress up.

**Comment by jackcrane** [comment]
Feature _should_ be finished, just awaiting review `:)`

File: templates/mobile/src/onboarding/slides/TagGroups/index.js (hunk 3 above)
Comment (dsilva609) [reply templates/mobile/src/onboarding/slides/TagGroups/index.js:4]: @jackcrane is this export still needed?

**Comment by dsilva609** [comment]
fixed a bad import and gave it a test and it's looking good 👍

**Review by dsilva609 — APPROVED** [review]

**Comment by camrun91** [comment]
@vinnyjth looks like this is waiting on @calebpanza to approve the changes it looks like they are good to go though. Do we want to force merge?

**Comment by vinnyjth** [comment]
@camrun91 I think we need to hold off on this one - we need to make sure it isn't going to show up unless there are tag groups configured. 

Can we update this PR to remove https://github.com/ApollosProject/apollos-apps/pull/2799/files#diff-d6323377510f7dd23611f7fee6055176da929eca9164e25e890cfe4633b5e937R68 these lines - so the screen doesn't show up at all, then do a follow up PR that actually adds this screen to onboarding _if_ there are tag groups?

**Comment by camrun91** [comment]
Ah I got it I didnt think about there not being tag groups.

**Comment by jackcrane** [comment]
Im not able to test it because the app is not launching for me (have been fighting dependencies). The entire feature returns null when there are no tags returned by the API, which should (in theory) not display the page at all and allow onboarding to flow naturally

**Comment by vinnyjth** [comment]
> The entire feature returns null when there are no tags returned by the API, which should (in theory) not display the page at all and allow onboarding to flow naturally

💥 

In that case @cameron - can you test the above logic to confirm and then merge? 

There's now a tag group editor in retool

**Comment by jackcrane** [comment]
After finally getting the app to work and testing, my logic did not, in fact, work. Working on a fix.

**Comment by jackcrane** [comment]
Functionality is there, need to write tests.

**Comment by dsilva609** [comment]
> Functionality is there, need to write tests.

@jackcrane pushed a small fix for an incorrect variable, but otherwise everything checks out 👍

**Comment by vinnyjth** [comment]
@camrun91 can you share the error you are seeing?

**Comment by camrun91** [comment]
Screenshot 2023-01-30 at 4 18 04 PM
@vinnyjth

**Comment by dsilva609** [comment]
> @dsilva609 @calebpanza after uninstalling the app and trying to reinstall to sim from this branch I can not get it to install at all it looks like some of the project files changed in this PR. I dk if those are causing issues or not.

I'm seeing the same too now. Tried cleaning and rebuilding in xcode and it didn't help. Won't even launch now.
@jackcrane are you still able to get it to launch?

**Comment by dsilva609** [comment]
@camrun91 are you able to get it to launch by using the previous version of the updated `.pbxproj` file?

File: templates/mobile/ios/apolloschurchapp.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -18,6 +18,7 @@
 		4CA6A042251BC0BF00DD0D97 /* SplashScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 4CA6A041251BC0BF00DD0D97 /* SplashScreen.storyboard */; };
 		578FFEC1D7D54C9B871A9E5B /* Inter-UI-BlackItalic.otf in Resources */ = {isa = PBXBuildFile; fileRef = 5A5A2BC2998547F4816B82E4 /* Inter-UI-BlackItalic.otf */; };
 		68D1351E80AA4A2C96288FD9 /* Inter-UI-Black.otf in Resources */ = {isa = PBXBuildFile; fileRef = EC21254B449443F99016EBFD /* Inter-UI-Black.otf */; };
+		6ABB80E329872BF5002D3938 /* NotificationService.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF49AE862649A852001C7B78 /* NotificationService.swift */; };
```
Comment (dsilva609) [inline templates/mobile/ios/apolloschurchapp.xcodeproj/project.pbxproj:21]: @jackcrane These updates seem to be causing the install to fail

**Comment by camrun91** [comment]
@dsilva609 yes matching the one that is in master fixes it. I am now trying to get the screen to show up.

File: templates/mobile/ios/apolloschurchapp.xcodeproj/project.pbxproj (hunk 6 above)
Comment (camrun91) [reply templates/mobile/ios/apolloschurchapp.xcodeproj/project.pbxproj]: @jackcrane @dsilva609 I just pushed changes that fix this. Jack if this is something that is needed for somereason please let us know.

**Comment by camrun91** [comment]
I can not get this to show up for some reason.

**Comment by camrun91** [comment]
OK I finally got this to work. I had to do some logging and it finally showed up when the sim reloaded in the middle of onboarding. I would love for someone else to validate that this works with out having to trouble shoot.

**Comment by jackcrane** [comment]
Those seem to have been leftovers from me attempting to get the app to launch. They can be reverted fine :)

**Comment by calebpanza** [comment]
@jackcrane could you also revert the addition of the `IDECheck` file? The rest of this is looking good to me

**Comment by jackcrane** [comment]
I will delete the extraneous file and re-test everything 👍

**Pull request merged**